src/components/App.tsx

```diff
@@ -2650,11 +2662,16 @@ class App extends React.Component<AppProps, AppState> {
         y,
       });
     } else if (this.state.elementType === "freedraw") {
-      this.handleFreeDrawElementOnPointerDown(
-        event,
-        this.state.elementType,
-        pointerDownState,
-      );
+      if (
+        (this.state.penLocked && event.pointerType === "pen") ||
+        !this.state.penLocked
```

Comment:
```suggestion
        (!this.state.penLocked || event.pointerType !== "touch")
```

Reply:
I don't understand your suggestion.
Why pointerType !== "touch" ?
what about pointerType === "mouse"?

I find my version easier to read and more clear... but then again, I am not sure I understand the logic you are proposing.

Reply:
Same as yours except I also allow mouse when in pen mode